funcs_utils.php

```diff
@@ -0,0 +1,239 @@
+        error("Output was: " . $process->getErrorOutput());
+    }
+    return trim($process->getOutput());
+}
+
+/**
+ * Returns a object used to output to the console
+ */
+function io(): SymfonyStyle
+{
+    global $IN, $OUT;
+    return new SymfonyStyle($IN ?: new ArgvInput(), $OUT ?: new NullOutput);
+}
+
+/**
+ * Removes a directory
+ */
+function remove_dir($dirname)
+{
+    if (!file_exists(($dirname))) {
+        return;
+    }
+    info("Removing $dirname");
+    shell_exec("rm -rf $dirname");
+}
+
+/**
+ * Validates the users system is ready to run the script
+ */
+function validate_system()
+{
+    if (!github_token()) {
+        error('Could not get github token - set MS_GITHUB_TOKEN environment variable');
+    }
+}
+
+/**
+ * Reads MS_GITHUB_TOKEN environment variable
+ */
+function github_token()
+{
+    return getenv('MS_GITHUB_TOKEN') ?: '';
+}
+
+/**
+ * Makes a request to the github API
+ */
+function github_api($url, $data = [])
+{
+    $token = github_token();
+    $jsonStr = empty($data) ? '' : json_encode($data, JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);
+    $ch = curl_init($url);
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+    curl_setopt($ch, CURLOPT_POST, !empty($data));
+    curl_setopt($ch, CURLOPT_HTTPHEADER, [
+        'User-Agent: silverstripe-module-standardiser',
+        'Accept: application/vnd.github+json',
+        "Authorization: Bearer $token",
+        'X-GitHub-Api-Version: 2022-11-28'
+    ]);
+    if ($jsonStr) {
+        curl_setopt($ch, CURLOPT_POSTFIELDS, $jsonStr);
+    }
+    $response = curl_exec($ch);
+    $httpcode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+    curl_close($ch);
+    if ($httpcode >= 300) {
+        warning("HTTP code $httpcode returned from GitHub API");
+        warning($response);
+        error("Failure calling github api: $url");
+    }
+    return json_decode($response, true);
+}
+
+/**
+ * Outputs a list of PRs created
+ * Prefixed with a dash so that it's easy to copy and paste into a parent github issue
+ */
+function output_prs_created()
+{
+    global $PRS_CREATED;
+    $io = io();
+    $io->writeln('');
+    $io->writeln('Pull requests created:');
+    foreach ($PRS_CREATED as $pr) {
+        $io->writeln("- $pr");
+    }
+    $io->writeln('');
+}
+
+/**
+ * Ouputs a list of repos that that had PRs created
+ * This is intended to be used when there was an error with a run (probably a secondary rate limit) and then
+ * copy pasted into the --exclude option for the next run
+ */
+function output_repos_with_prs_created()
+{
+    global $REPOS_WITH_PRS_CREATED;
+    $io = io();
+    $io->writeln('');
+    $io->writeln('Repos with pull requests created (add to --exclude if you need to re-run):');
+    $io->writeln(implode(',', $REPOS_WITH_PRS_CREATED));
+    $io->writeln('');
+}
+
+/**
+ * Works out which branch in a module to checkout before running scripts on it
+ *
+ * Assumption is being made that the modules default branch is the next-minor branch
```

Comment:
Can't rely on this assumption e.g. symbiote/silverstripe-advanced-workflow is wrong

Will need to do some composer.json detection similar to gha-merge-ups